src/lem/eval.rs

```diff
@@ -957,11 +968,11 @@ fn apply_cont() -> Func {
                                     Expr::U64 => {
                                         let val = add(evaled_arg, result);
                                         let not_overflow = lt(val, size_u64);
-                                        if not_overflow == zero {
-                                            let val = sub(val, size_u64);
+                                        if not_overflow {
```

Comment:
Although only tangentially related to this work, let me still note that this depends on the args having been correctly tagged. At one point, we discussed enforcing a check (constructing from 64 bits) on the args. Has this been implemented somewhere? I didn't see it. If not, let's make an issue for it.

Reply:
I'm going to create an issue for it so we can answer this question (and/or act upon it) later